staff/lab/mod-printer

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python3
+"""Remove or add printers to/from CUPS classes."""
+import argparse
+import subprocess
+import sys
+
+
+def modify_printer(args):
+    """Perform internal call to CUPS CLI to modify class(es)."""
+    # Determine which CUPS classes are available
+    cups_classes = subprocess.check_output(["lpstat", "-p"]).decode("utf-8").split("\n")
+    # Remove empty trailing newline
+    cups_classes.remove("")
+    for i in range(len(cups_classes)):
+        clazz = cups_classes[i]
+        # Output is of the form: printer <printername> is idle. <etc>
+        clazz = clazz[len("printer "):]
+        clazz = clazz[:clazz.index(" is ")]
+        cups_classes[i] = clazz
+
+    if args.verbose:
+        print("Found CUPS classes: " + str(cups_classes))
+    
+    args_clazz = getattr(args, "class")
+    if args_clazz and args_clazz in cups_classes:
+        cups_classes = [args_clazz]
+        if args.verbose:
+            print("Overriding CUPS classes with parameter: " + str(args_clazz))
+    
+    # Perform pre-call logging and input processing
+    printer = args.printer.lower()
+    if args.verbose:
+        print("Printer to perform action on: " + printer)
+    action_flag = "-r" if args.action == "remove" else "-c"
+    if args.verbose:
+        print("Action to be performed (and flag): " + args.action, "(" + action_flag + ")")
+        
+    # Call lpadmin command to perform addition/removal
+    for clazz in cups_classes:
+        action_cmd = subprocess.Popen(["lpadmin", "-p", printer + "-" + clazz, action_flag, clazz])
```

Comment:
Again, use `subprocess.run` here. You can use the `check=True` option to raise on an error code.